test/multiverse/suites/stripe/stripe_instrumentation_test.rb

```diff
@@ -0,0 +1,160 @@
+          in_transaction do |txn|
+            Stripe::Customer.list({limit: 3}) # Start a Stripe event
+            stripe_segment = txn.segments.detect { |s| s.name == 'Stripe/v1/customers get' }
+
+            assert(stripe_segment)
+            stripe_attributes = stripe_segment.attributes.agent_attributes_for(NewRelic::Agent::AttributeFilter::DST_SPAN_EVENTS)
+
+            assert_equal('meow', stripe_attributes['stripe_user_data_cat'])
+            assert_equal('woof', stripe_attributes['stripe_user_data_dog'])
+          end
+        end
+      end
+    end
+  end
+
+  def test_agent_collects_select_user_data_attributes
+    Stripe::Instrumentation.subscribe(:request_begin) do |events|
+      events.user_data[:frog] = 'ribbit'
+      events.user_data[:sheep] = 'baa'
+      events.user_data[:cow] = 'moo'
+    end
+
+    with_config(:'stripe.user_data.include' => 'frog, sheep') do
+      Stripe::StripeClient.stub(:default_connection_manager, @connection) do
+        @connection.stub(:execute_request, @response) do
+          in_transaction do |txn|
+            Stripe::Customer.list({limit: 3}) # Start a Stripe event
+            stripe_segment = txn.segments.detect { |s| s.name == 'Stripe/v1/customers get' }
+
+            assert(stripe_segment)
+            stripe_attributes = stripe_segment.attributes.agent_attributes_for(NewRelic::Agent::AttributeFilter::DST_SPAN_EVENTS)
+
+            assert_equal('ribbit', stripe_attributes['stripe_user_data_frog'])
+            assert_equal('baa', stripe_attributes['stripe_user_data_sheep'])
+            assert_nil(stripe_attributes['stripe_user_data_cow'])
+          end
+        end
+      end
+    end
+  end
+
+  def test_agent_ignores_user_data_attributes
+    Stripe::Instrumentation.subscribe(:request_begin) do |events|
+      events.user_data[:bird] = 'tweet'
+    end
+
+    with_config(:'stripe.user_data.exclude' => 'bird') do
+      Stripe::StripeClient.stub(:default_connection_manager, @connection) do
+        @connection.stub(:execute_request, @response) do
+          in_transaction do |txn|
+            Stripe::Customer.list({limit: 3})
+            stripe_segment = txn.segments.detect { |s| s.name == 'Stripe/v1/customers get' }
+
+            assert(stripe_segment)
+            stripe_attributes = stripe_segment.attributes.agent_attributes_for(NewRelic::Agent::AttributeFilter::DST_SPAN_EVENTS)
+
+            assert_nil(stripe_attributes['stripe_user_data_bird'])
+          end
+        end
+      end
+    end
+  end
+
+  def test_start_when_not_traced
+    Stripe::StripeClient.stub(:default_connection_manager, @connection) do
+      @connection.stub(:execute_request, @response) do
+        NewRelic::Agent::Tracer.state.stub(:is_execution_traced?, false) do
+          in_transaction do |txn|
+            Stripe::Customer.list({limit: 3})
+            stripe_segment = txn.segments.detect { |s| s.name == 'Stripe/v1/customers get' }
+
+            assert_empty txn.segments
```

Comment:
There's always a single "dummy" segment when you use `in_transaction`. You should be able to do:

```ruby
refute stripe_segment
```

but you don't want the transaction's segment array itself to be empty, as it'll always have at least 1 entry.

Reply:
That makes sense. I've updated what we're looking for https://github.com/newrelic/newrelic-ruby-agent/pull/2180/commits/63014dd5d7236c3fb38390840d7cfc0ff0ed70a4